quickwit/quickwit-metastore/src/metastore/file_backed_metastore/mod.rs

```diff
@@ -642,6 +650,84 @@ impl Metastore for FileBackedMetastore {
         check_indexes_states_exist(self.storage.clone()).await
     }
 
+    // Shard API
+
+    async fn open_shards(&self, request: OpenShardsRequest) -> MetastoreResult<OpenShardsResponse> {
+        let mut subresponses = Vec::with_capacity(request.subrequests.len());
+
+        for subrequest in request.subrequests {
+            let index_uid: IndexUid = subrequest.index_uid.clone().into();
+            let subresponse = self
+                .mutate(index_uid, |index| index.open_shards(subrequest))
+                .await?;
+            subresponses.push(subresponse);
+        }
+        let response = OpenShardsResponse { subresponses };
+        Ok(response)
+    }
+
+    async fn close_shards(
```

Comment:
Can we add in the documentation of those requests that they are not executed atomically at all?

(this is ok, but I'd feel better if it was documented)